.github/workflows/review_and_testing.yml

```diff
@@ -1,10 +1,33 @@
-name: testing
+name: Review & Test
+
+permissions:
+  contents: read
+  pull-requests: write
 
 on:
   pull_request:
     branches: [main]
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This workflow is triggered on pull requests targeting the `main` branch. This is a common practice to ensure that all changes proposed to the main codebase are reviewed and tested before they are merged.

```diff
  on:
    pull_request:
      branches: [main]
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->